synapse_invite_checker/invite_checker.py

```diff
@@ -1132,30 +1149,32 @@ async def room_scan(self) -> None:
 
                 # only shut down rooms that only have EPA hosts in them
                 if await self.have_all_pro_hosts_left(room_id):
-                    last_user_left_timestamp: int | None = (
-                        await self.get_timestamp_from_eligible_events_for_epa_room_purge(
-                            room_id
-                        )
+                    epa_room_result = await self.define_criteria_to_kick_users_epa_mode(
+                        room_id
                     )
+                    current_time = self.api._hs.get_clock().time_msec()
+
+                    if (
+                        epa_room_result.should_kick_because_leave(current_time)
+                        or epa_room_result.should_kick_because_invite(current_time)
+                        or epa_room_result.should_kick_because_no_activity_since_creation(
+                            current_time
+                        )
```

Comment:
this logic seems backwards, it should be like this:

kick if create event, leave event and invite are older than their respective limits. You can fold create and leave in together, invite should only be considered if there is an invite.